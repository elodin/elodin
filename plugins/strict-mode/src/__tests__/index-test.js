import { traverse, parse } from '@elodin/core'

import strictMode from '../index'

describe('Strict mode', () => {
  it('should correctly throw on variables', () => {
    const file = `style Button { paddingLeft: 10 }`

    const ast = parse(file).ast

    expect(traverse(ast, [strictMode()])).toMatchSnapshot()
  })
})
