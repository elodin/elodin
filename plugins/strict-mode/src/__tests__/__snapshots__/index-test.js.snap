// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Strict mode should correctly throw on variables 1`] = `
Object {
  "body": Array [
    Object {
      "body": Array [
        Object {
          "comments": Array [],
          "dynamic": false,
          "property": "paddingLeft",
          "raw": false,
          "type": "Declaration",
          "value": Object {
            "negative": false,
            "type": "Integer",
            "value": 10,
          },
        },
      ],
      "comments": Array [],
      "name": "Button",
      "type": "Style",
    },
  ],
  "type": "File",
}
`;
